abis_mapping/plugins/list.py

```diff
@@ -0,0 +1,103 @@
+"""Provides custom frictionless list plugin for the package"""
+
+
+# Third-Party
+import frictionless
+
+# Typing
+from typing import Any, Optional
+
+
+class ListPlugin(frictionless.Plugin):
+    """Custom List Plugin"""
+
+    # Class Attributes
+    code = "list"
+    status = "stable"
+
+    def create_type(self, field: frictionless.Field) -> Optional[frictionless.Type]:
+        """Create type from this plugin
+
+        Args:
+            field (frictionless.Field): Corresponding field.
+
+        Returns:
+            Optional[frictionless.Type]: Possible type from this plugin.
+        """
+        # Check for our type
+        if field.type == ListType.code:
+            # Return
+            return ListType(field)
+
+        # Not our type
+        return None
+
+
+class ListType(frictionless.Type):
+    """Custom List Type Implementation."""
+
+    # Class Attributes
+    code = "list"
+    builtin = False
+    constraints = [
+        "required",
+        "minLength",
+        "maxLength",
+        # "pattern",  # TODO -> Check whether this works
+        # "enum",  # TODO -> Check whether this works
+    ]
+
+    # Delimiter
+    delimiter = " "
```

Comment:
mmm, `space` as a delimiter will allow only single word values. I'd rather see a comma as delimiter, unless frictionless already support `,` as delimiter (see comments below)

Reply:
I agree, there is still a lot of discussion around this `associatedSequences` field that uses this custom list plugin.

It was decided that `json` notation for an array (e.g., `["A", "B", "C"]`) was not user friendly for someone filling out the template.

Originally, this field was meant to only be URIs, so space separator made sense. Then we found out that WA BIO can't guarantee that they will be URIs - so we will have to be able to handle anything.

TLDR: See: BDR-628 comments